reasoner/Reasoner.java

```diff
@@ -80,14 +81,16 @@ public class Reasoner {
     private final ControllerRegistry controllerRegistry;
     private final ExplainablesManager explainablesManager;
     private final ReasonerPlanner planner;
+    private final ReasonerPerfCounters perfCounters;
 
     public Reasoner(ConceptManager conceptMgr, LogicManager logicMgr,
                     TraversalEngine traversalEng, Context.Transaction context) {
         this.conceptMgr = conceptMgr;
         this.traversalEng = traversalEng;
         this.logicMgr = logicMgr;
-        this.planner = ReasonerPlanner.create(traversalEng, conceptMgr, logicMgr, context.options().explain());
-        this.controllerRegistry = new ControllerRegistry(actor(), traversalEng, conceptMgr, logicMgr, planner, context);
+        this.perfCounters = new ReasonerPerfCounters(context.options().infer() && context.options().reasonerPerfCounters());
```

Comment:
Enabled if infer is on & reasonerPerfCounters is enabled, which is set through config file.